server/src/main/scala/org/http4s/server/middleware/URITranslation.scala

```diff
@@ -2,21 +2,33 @@ package org.http4s
 package server
 package middleware
 
-import cats.Functor
+import cats.{Applicative, Functor}
+import cats.data.{Kleisli, OptionT}
 
 object URITranslation {
-  def translateRoot[F[_]: Functor](prefix: String)(service: HttpService[F]): HttpService[F] = {
-    val newCaret = prefix match {
-      case "/" => 0
-      case x if x.startsWith("/") => x.length
-      case x => x.length + 1
-    }
+  def translateRoot[F[_]: Applicative](prefix: String)(service: HttpService[F]): HttpService[F] =
```

Comment:
This changes the function signature, so isn't binary compatible

Reply:
Hmm, yes, that's true.
I'm not sure if this can be solved without being to signal a not-matched via `OptionT.none`.

Reply:
@rossabaker could you take a look at this? Should I retarget master and just let this be a known bug in 0.18?

Reply:
Our type is called `Uri`, but this object is called `URI`.  That's inconsistent.  We could deprecate thid old buggy one, and use this new implementation as `UriTranslation` in release-0.18.x.  Or `TranslateUri`.

Reply:
:+1: